### ISSUE TITLE
Configure View Entry Title for Relation field with User (from admin) not working

### ISSUE THREAD

**Issue by MIDimitrovS** [strapi/strapi#18072]
### Required System information
- Node.js version: 18.12.1
- NPM version: 8.19.2
- Strapi version: 4.13.6
- Database: Postgre
- Operating system: Windows 10
- Is your project Javascript or Typescript: Javascript

### Describe the bug
When I configure the entry title for any of the relation field with User (from admin), the Content Manager is not updated

### Steps to reproduce the behavior

1. Add a field that is relation with the User (from admin) to any content type
2. Configure the view for this content type
3. Attempt to create an item
4. The Relation Field input only shows the id of the user
5. The record is shown correct in the content manager

### Expected behavior

I expect to see the field input show the User's field value of the configured view in my case firstname

### Screenshots
List of Users
![users](https://github.com/strapi/strapi/assets/140058571/d951efb5-22e4-4a3b-b0ec-bb9eaf833563)
Creation of collection type 
![Enity](https://github.com/strapi/strapi/assets/140058571/e4369725-1490-4d74-bd3d-98c7f4fe562a)
Configure the view
![view](https://github.com/strapi/strapi/assets/140058571/c215fa95-cd5b-4815-89ff-afa5256a646e)
Creation of record
![record](https://github.com/strapi/strapi/assets/140058571/ff64f8b4-d39b-460e-99cb-2e73093818d3)
![manager](https://github.com/strapi/strapi/assets/140058571/31cff482-c858-4521-808e-ed1b0fd5536b)

### Additional context
It is supposed to be fixed in https://github.com/strapi/strapi/issues/17561 and the problem should not exist in v4.13.1 and above.

**Comment by derrickmehaffy** [comment]
Most likely related to the security fixes

**Comment by derrickmehaffy** [comment]
Confirmed on a recent experimental (based on 4.13.6):
![image](https://github.com/strapi/strapi/assets/8593673/1437c2de-05f0-4a62-b5a7-8f71f11ea817)
![image](https://github.com/strapi/strapi/assets/8593673/5e55cb22-65cb-4ab4-ae35-17ee923d583f)
![image](https://github.com/strapi/strapi/assets/8593673/d049ea51-5e2e-450f-b1e6-3e7272acb645)

**Comment by arpadgabor** [comment]
Seems like the query for the relation select box only returns id's.
![image](https://github.com/strapi/strapi/assets/9270494/e05f6900-e222-4e6c-b0dc-112b1eb19196)

The related field looks also empty in the content manager response. (`author` being the field with admin relation)
![image](https://github.com/strapi/strapi/assets/9270494/449aa384-7ec9-43c8-a848-3452badec48c)

**Comment by derrickmehaffy** [comment]
> Seems like the query for the relation select box only returns id's.
> ![image](https://github.com/strapi/strapi/assets/9270494/e05f6900-e222-4e6c-b0dc-112b1eb19196)
> 
> The related field looks also empty in the content manager response. (`author` being the field with admin relation)
> ![image](https://github.com/strapi/strapi/assets/9270494/449aa384-7ec9-43c8-a848-3452badec48c)
> 
> Edit: 4.13.3

That's normal here as we fetch the relations differently and not in one massive request, there should be another that pulls the relationship data.

**Comment by arpadgabor** [comment]
@derrickmehaffy I think the first screenshot represents that? My relation is called `authors` and that was the resource it was fetching but only returned the id's.

**Comment by derrickmehaffy** [comment]
Can you show the request information (URL, request headers)? Feel free to block out any private information.

**Comment by arpadgabor** [comment]
Yes, I created a test content type. My relation field is called `admin_users` and the content type is `test`.

Request:
```http
GET /content-manager/relations/api::test.test/1/admin_users?pageSize=5&page=1 HTTP/1.1
Host: localhost:1337
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
Authorization: <omitted for brevity>
Connection: keep-alive
Cookie: <omitted for brevity>
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-origin
```

Response:
```http
HTTP/1.1 200 OK
Content-Security-Policy: <omitted>
Referrer-Policy: no-referrer
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Content-Type-Options: nosniff
X-DNS-Prefetch-Control: off
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
Vary: Origin
Content-Type: application/json; charset=utf-8
X-Powered-By: Strapi <strapi.io>
Content-Length: 83
Date: Fri, 15 Sep 2023 19:33:34 GMT
Connection: keep-alive
Keep-Alive: timeout=5

BODY

{
  "results": [{ "id":1 }],
  "pagination": { "page":1, "pageSize":5, "pageCount":1, "total":1 }
}
```

**Comment by derrickmehaffy** [comment]
Indeed yup, it seems to be only grabbing the ID and not including the field for the actual display field.

**Comment by nicoXXXXXtitan** [comment]
Hello

For your information, I have a problem similar to yours on this issue. 
Have a nice day!

**Comment by lordvcs** [comment]
I am facing the same issue too

**Comment by KaloyanTsvetkov** [comment]
Is there a release date for the fix?

**Comment by alissawix** [comment]
I just want to add that it's the same for a relation to any collection
strapi version 4.15.4

**Comment by Charles-Abasimfon** [comment]
Same issue here too.

**Comment by kareraisu** [comment]
Hello! Thanks to the reporter and to the devs for all the hard work.
I am experiencing this as well in v4.15.5.
Cheers!

**Comment by thomasbnt** [comment]
Hello! Same issue on `v4.19.1` with Node `v18.15.0`

**Comment by derrickmehaffy** [comment]
Locking this as we are getting a lot of unhelpful messages, we are aware and will look at this when we have the time.
